query/src/sql/sql_parser.rs

```diff
@@ -73,12 +75,26 @@ impl<'a> DfParser<'a> {
     }
 
     /// Parse a SQL statement and produce a set of statements with dialect
-    pub fn parse_sql(sql: &'a str) -> Result<(Vec<DfStatement<'a>>, Vec<DfHint>), ErrorCode> {
-        let dialect = &GenericDialect {};
-        let start = Instant::now();
-        let result = DfParser::parse_sql_with_dialect(sql, dialect)?;
-        histogram!(super::metrics::METRIC_PARSER_USEDTIME, start.elapsed());
-        Ok(result)
+    pub fn parse_sql(
+        sql: &'a str,
+        typ: SessionType,
+    ) -> Result<(Vec<DfStatement<'a>>, Vec<DfHint>), ErrorCode> {
+        match typ {
+            SessionType::MySQL => {
+                let dialect = &MySqlDialect {};
```

Comment:
Looks like there are some duplicate codes.
How about
```rust
let dialect = match ...
```

Reply:
Sessiontype has different tpye .
I can not set a val with diff type